posthog/urls.py

```diff
@@ -80,7 +80,7 @@ def social_create_user(strategy: DjangoStrategy, details, backend, request, user
     if user:
         return {"is_new": False}
     user_email = details["email"][0] if isinstance(details["email"], (list, tuple)) else details["email"]
-    user_name = details["fullname"]
+    user_name = details["fullname"] or ""
```

Comment:
Hm, I don't think this changes anything, as at worst there's no `fullname` attr on `details`, in which case there's still a `KeyError`, or the attr can be an empty string, in which case it's `or`'d to… an empty string too.

Reply:
Hmm I definitely didn't look into it too deeply but I had a hunch this would have been `None`. It's not throwing a `KeyError`, but rather a `ValidationError` later down the line. And we create users with empty string first names in `bootstrap` for example

https://sentry.io/organizations/posthog/issues/2358536963/?project=1899813&query=is%3Aunresolved&statsPeriod=14d

Reply:
Right, so in this case Sentry context shows that it already is not null but an empty string (which to Django is "blank"). So can't use an empty string as a fallback here. Take a look now though – using `details["username"]` as fallback. This will fix the problem for GitHub/GitLab, e.g. if I tried to sign up with GH auth having no real name set on my GH profile, I'd experience this `ValidationError` before. With `username` as fallback, I'd be able to sign up, it's just that my name would be "Twixes" in PostHog – doesn't look super professional, but at least I can use the app.

Reply:
That's true, change makes sense to me